matchzoo/contrib/models/esim.py

```diff
@@ -0,0 +1,238 @@
+"""ESIM model."""
+import keras
+import tensorflow as tf
+import keras.backend as K
+
+import matchzoo as mz
+from matchzoo.engine.param import Param
+from matchzoo.engine.base_model import BaseModel
+from matchzoo.engine.param_table import ParamTable
+
+
+class ESIM(BaseModel):
+    """
+    ESIM model.
+
+    Examples:
+        >>> model = ESIM()
+        >>> task = classification_task = mz.tasks.Classification(num_classes=2)
+        >>> model.params['task'] = task
+        >>> model.params['input_shapes'] = [(20, ), (40, )]
+        >>> model.params['lstm_dim'] = 300
+        >>> model.params['mlp_num_units'] = 300
+        >>> model.params['embedding_input_dim'] =  5000
+        >>> model.params['embedding_output_dim'] =  10
+        >>> model.params['embedding_trainable'] = False
+        >>> model.params['mlp_num_layers'] = 0
+        >>> model.params['mlp_num_fan_out'] = 300
+        >>> model.params['mlp_activation_func'] = 'tanh'
+        >>> model.params['mask_value'] = 0
+        >>> model.params['dropout_rate'] = 0.5
+        >>> model.params['optimizer'] = keras.optimizers.Adam(lr=4e-4)
+        >>> model.guess_and_fill_missing_params()
+        >>> model.build()
+    """
+
+    @classmethod
+    def get_default_params(cls) -> ParamTable:
+        """Get default parameters."""
+        params = super().get_default_params(with_embedding=True,
+                                            with_multi_layer_perceptron=True)
+
+        params.add(Param(
+            name='dropout_rate',
+            value=0.5,
+            desc="The dropout rate for all fully-connected layer"
+        ))
+
+        params.add(Param(
+            name='lstm_dim',
+            value=300,
+            desc="The dimension of LSTM layer."
+        ))
+
+        params.add(Param(
+            name='mask_value',
+            value=0,
+            desc="The value would be regarded as pad"
+        ))
+
+        return params
+
+    def _make_inputs(self) -> list:
+        input_left = keras.layers.Input(
+            name='text_left',
+            shape=self._params['input_shapes'][0]
+        )
+        input_right = keras.layers.Input(
+            name='text_right',
+            shape=self._params['input_shapes'][1]
+        )
+        return [input_left, input_right]
+
+    def _make_embedding_layer(self,
+                              name: str = 'embedding') -> keras.layers.Layer:
+        """
+        Overwrite the :meth:`BaseModel._make_embedding_layer` to allow
+        specifiying mask_zero.
+
+        :param name: name for embedding layer, default 'embedding'
+        """
+        return keras.layers.Embedding(
+            self._params['embedding_input_dim'],
+            self._params['embedding_output_dim'],
+            trainable=self._params['embedding_trainable'],
+            mask_zero=True,
```

Comment:
Only slightly different from the default one (`mask_zero`), we could update the base method to take `mask_zero` as a parameter or something.

Reply:
shall I send another pr for that and remove this overriding function? or just leave it for now

Reply:
#688 should solve the problem, you may cherrypick it now or rebase onto it once it's merged. Also, I added you as one of the reviewers, so if it works, approve it.